scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java

```diff
@@ -977,5 +981,44 @@ TaskUsage toTaskUsage(JobIdImpl jobId) {
             getExecutionDuration(), getParallelEnvironment() == null ? 1 : getParallelEnvironment()
                     .getNodesNumber());
     }
+    
 
+    TaskInfoImpl createTaskInfo(JobIdImpl jobId) {
+        TaskId taskId = TaskIdImpl.createTaskId(jobId, getTaskName(), getId().getTaskId());
+        TaskInfoImpl taskInfo = new TaskInfoImpl();
+        taskInfo.setTaskId(taskId);
+        taskInfo.setStatus(getTaskStatus());
+        taskInfo.setStartTime(getStartTime());
+        taskInfo.setProgress(0);
+        taskInfo.setNumberOfExecutionOnFailureLeft(getNumberOfExecutionOnFailureLeft());
+        taskInfo.setNumberOfExecutionLeft(getNumberOfExecutionLeft());
+        taskInfo.setJobInfo(getJobData().toJobInfo());
+        taskInfo.setJobId(jobId);
+        taskInfo.setFinishedTime(getFinishedTime());
+        taskInfo.setExecutionHostName(getExecutionHostName());
+        taskInfo.setExecutionDuration(getExecutionDuration());
+        return taskInfo;
+    }
+    
+    public TaskInfo toTaskInfo() {
+        JobIdImpl jobId = new JobIdImpl(getJobData().getId(), getJobData().getJobName());
+        TaskInfoImpl taskInfo = createTaskInfo(jobId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "taskInfo". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)